src/platform_layers/linux_platform_layer/src/linux_adu_core_impl.cpp

```diff
@@ -186,6 +186,20 @@ ADUC_Result LinuxPlatformLayer::Download(const char* workflowId, const char* upd
 
         extendedResultCode = MAKE_ADUC_DELIVERY_OPTIMIZATION_EXTENDEDRESULTCODE(doErrorCode);
     }
+    catch (const std::exception& e)
+    {
+        Log_Error("DP download failed with an unhandled std exception: %s", e.what());
+
+        resultCode = ADUC_DownloadResult_Failure;
+        extendedResultCode = ADUC_ERC_NOTRECOVERABLE;
```

Comment:
Can you check if we should try to get last error (e.g. `errno`) to see if we can return a better error code instead of  ADUC_ERC_NOTRECOVERABLE?

Also, going forward, we should try report 'resultDetails' property w/ any error, if possible.

In this case, e.what() would be very useful for diagnostic.

Reply:
Do we have a resultDetails field? 


---
In reply to: [603721077](https://github.com/Azure/iot-hub-device-update/pull/28#discussion_r603721077) [](ancestors = 603721077)